nixos/lib/test-driver/test_driver/machine.py

```diff
@@ -154,13 +154,12 @@ def cmd(
         # qemu options
         qemu_opts = ""
         qemu_opts += (
-            ""
-            if allow_reboot
-            else " -no-reboot"
             " -device virtio-serial"
             " -device virtconsole,chardev=shell"
             " -device virtio-rng-pci"
             " -serial stdio"
+            if allow_reboot
+            else " -no-reboot"
         )
```

Comment:
this whole block might be a bit better to read if it was structured like this:

```python
if allow_reboot:
    qemu_opts += (
        " -device virtio-serial"
        " -device virtconsole,chardev=shell"
        " -device virtio-rng-pci"
        " -serial stdio"
    )
else:
    qemu_opts += " -no-reboot"
```

In this format, people with less python experience also wouldn't have to ask themselves if in the `True`-case only the last string before the `if` is used or all of them.

Reply:
But also, i do not understand why all these flags shall be removed in the no-reboot case...

Reply:
Apologies, I'm not very savvy with Python syntax. Fixed that!